dnsDisco_test: refactor structs to better reading

arxdsilva opened the pull request.

Commit:
dnsDisco_test: refactor structs to better reading

rafaeljusto commented:
Hi @ArxdSilva, thanks for the PR!

But I don't see an improvement on reading here. As the table driven test structs contains common fields (except for expectedError) embedding appears to be an approach, but then you see that adding a scenario becomes more verbose as you need to identify the internal type. Also, why did you remove so many blank lines? Saving bytes? 😃 

Maybe we could just move out the scenarios from the functions to make them smaller and clean. And this would also follow the reference of the [Go wiki page](https://github.com/golang/go/wiki/TableDrivenTests).

Commit:
dnsdisco_test: moved test cases

blank lines are now back

Commit:
adds contributor

arxdsilva commented:
Hi @rafaeljusto , changed back things as they were, modified the scenarios names into a bit more intuitive ones as they'll be out of func - hope you like It. 

BTW do you know why your  `-repotoken` is asking for a parameter? Your previous builds didn't pass It too and It had no problem.

rafaeljusto commented:
That's great! Thank you.

The problem related to the `-repotoken` is, according to [Travis Docs](https://docs.travis-ci.com/user/environment-variables/#Defining-encrypted-variables-in-.travis.yml), because of the encrypted global variable that is not available in pull requests.

> Encrypted environment variables are not available to pull requests from forks due to the security risk of exposing such information to unknown code.

I will take a better look on this to make it work with PRs.
